Assets/Xbox Live/Scripts/UserProfile.cs

```diff
@@ -77,39 +92,32 @@ public void Start()
         }
         this.Refresh();
 
-        SocialManagerComponent.Instance.EventProcessed += SocialManagerEventProcessed;
-        Microsoft.Xbox.Services.XboxLiveUser.SignOutCompleted += XboxLiveUserOnSignOutCompleted;
-
-        if (XboxLiveUserManager.Instance.SingleUserModeEnabled)
+        try
         {
-            if (XboxLiveUserManager.Instance.UserForSingleUserMode == null)
+            SocialManagerComponent.Instance.EventProcessed += SocialManagerEventProcessed;
+            SignInManager.Instance.OnPlayerSignOut(this.PlayerNumber, this.XboxLiveUserOnSignOutCompleted);
+            SignInManager.Instance.OnPlayerSignIn(this.PlayerNumber, this.SetupForRefresh);
+            this.xboxLiveUser = SignInManager.Instance.GetUser(this.PlayerNumber);
```

Comment:
Why do you call GetUser here, when it is also done in the SetupForRefresh callback method?

Reply:
This is called when switching scenes. The user would have already been signed in from the previous scenes and so callbacks won't be happening. Callbacks are only called the moment the player signs in and signs out basically.

So in the case of scene switching, this will attempt to GetUser. And if it's not null, it will proceed with loading the profile.